elastic/beats
# Additional stats fields for Elasticsearch

## Proposed commit message

Adds `creation_date` and `tier_preference` fields for `elasticsearch.index` dataset.
This will be necessary for further development through https://github.com/elastic/integrations/pull/11656

## Checklist



- [x] ~I have made corresponding change to the default configuration files~ N/A

Regarding the [documentation](https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-metricset-elasticsearch-index.html), the example document is copied from the `data.json` file, accurately modified in this PR.

Another modification in the `integrations` repo will be required (for [this file](https://github.com/elastic/integrations/tree/main/packages/elasticsearch/docs))

## Disruptive User Impact

This "shouldn't" have an impact on end-users, this doesn't alter existing behavior but only adds 2 new fields that will be exposed in the gathered Elasticsearch monitoring stats.

## Author's Checklist

- [ ] Modify the integrations repository so that the documentation is accurate there: https://github.com/elastic/integrations/pull/11942
- [ ] Modify the Elasticsearch code for the built-in index template: https://github.com/elastic/elasticsearch/pull/117851

## How to test this PR locally

You can run the integration against any cluster (with `xpack` or otherwise) and check that the generated index stats documents have the two new fields:

- `creation_date`
- `tier_preference`

## Screenshots

<img width="715" alt="image" src="https://github.com/user-attachments/assets/f5fc3cce-596e-4e28-8411-6924058e5ac6">
